linters/export_delivery_data/join_parent_issues.py

```diff
@@ -0,0 +1,206 @@
+        # If the parent matches the desired type, return it
+        if IssueType(parent.issue_type) == type_wanted:
+            return parent
+        # If the parent doesn't have a its own parent, return None
+        if not parent.issue_parent:
+            return None
+        # Otherwise update the parent_url to "grandparent" and continue
+        parent_url = parent.issue_parent
+
+    # Return the URL of the parent deliverable (or None)
+    return None
+
+
+def flatten_issue_data(lookup: dict[str, IssueMetadata]) -> list[dict]:
+    """Flatten issue data and inherit data from parent epic an deliverable."""
+    result: list[dict] = []
+    for issue in lookup.values():
+        # If the issue is a deliverable or epic, move to the next one
+        if IssueType(issue.issue_type) in [IssueType.DELIVERABLE, IssueType.EPIC]:
+            continue
+
+        # Get the parent deliverable, if the issue has one
+        deliverable = get_parent_with_type(
+            child_url=issue.issue_url,
+            lookup=lookup,
+            type_wanted=IssueType.DELIVERABLE,
+        )
+        if deliverable:
+            # Set deliverable metadata
+            issue.deliverable_title = deliverable.issue_title
+            issue.deliverable_url = deliverable.issue_url
+            issue.pillar = deliverable.pillar
+            # Set quad metadata
+            issue.quad_id = deliverable.quad_id
+            issue.quad_name = deliverable.quad_name
+            issue.quad_start = deliverable.quad_start
+            issue.quad_end = deliverable.quad_end
+            issue.quad_length = deliverable.quad_length
+
+        # Get the parent epic, if the issue has one
+        epic = get_parent_with_type(
+            child_url=issue.issue_url,
+            lookup=lookup,
+            type_wanted=IssueType.EPIC,
+        )
+        if epic:
+            issue.epic_title = epic.issue_title
+            issue.epic_url = epic.issue_url
+
+        # Add the issue to the results
+        result.append(issue.__dict__)
+
+    # Return the results
+    return result
+
+
+def run_transformations(
+    sprint_file_in: str,
+    roadmap_file_in: str,
+    task_file_out: str,
+) -> None:
+    """Runs a transformation pipeline to transform issue data to the correct format."""
+    # Load sprint and roadmap data
+    sprint_data_in = load_json_file(sprint_file_in)
+    roadmap_data_in = load_json_file(roadmap_file_in)
+    # Populate a lookup table with this data
+    lookup = {}
+    lookup = populate_issue_lookup_table(lookup, roadmap_data_in)
+    lookup = populate_issue_lookup_table(lookup, sprint_data_in)
+    # flatten and write issue level data to output file
+    tasks_out = flatten_issue_data(lookup)
+    dump_to_json(task_file_out, tasks_out)
+
+
+if __name__ == "__main__":
+
+    # Create parser
+    parser = argparse.ArgumentParser(
+        prog="ProgramName",
+        description="What the program does",
+        epilog="Text at the bottom of help",
```

Comment:
Probably a good idea to put an actual description etc here :)

Reply:
Great catch! Updated that in this commit: [docs: Improve python script documentation](https://github.com/agilesix/simpler-grants-sandbox/pull/43/commits/e0d1d3c9517eb4abdb682a75ff4c72be824fdf56)